File: NSNotification-Demo/NSNotification-Demo/ViewController.m
//
//  ViewController.m
//  NSNotification-Demo
//
//  Created by kingcos on 2021/4/20.
//

#import "ViewController.h"
#import "NSNotificationTester.h"

@interface ViewController ()

@property (nonatomic, strong) NSNotificationTester *tester;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
//    [self.tester testObserveAllNotifications];
    [self.tester testObserveMultipleTimes];
//    [self.tester testRemoveMultipleTimes];
}

- (IBAction)addObserver:(id)sender {
    
}

- (IBAction)removeObserver:(id)sender {
    [self.tester testRemoveMultipleTimes];
}

- (NSNotificationTester *)tester {
    if (!_tester) {
        _tester = [[NSNotificationTester alloc] init];
    }
    
    return _tester;
}

@end

File: UIButton-demo/UIButton-demo/ViewController.swift
//
//  ViewController.swift
//  UIButton-demo
//
//  Created by kingcos on 2020/9/10.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var titleTopPicker: UIPickerView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
//        titleTopPicker.dataSource = self
    }
    
    
}

extension ViewController: UIPickerViewDataSource, UIPickerViewDelegate {
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        1
    }
    
    func pickerView(_ pickerView: UIPickerView,
                    numberOfRowsInComponent component: Int) -> Int {
        1
    }
    
    func pickerView(_ pickerView: UIPickerView,
                    titleForRow row: Int,
                    forComponent component: Int) -> String? {
        "1"
    }
}
